service/src/main/java/bio/terra/externalcreds/services/TokenProviderService.java

```diff
@@ -113,24 +104,6 @@ public Optional<String> getProviderAccessToken(
                                 + "Please go to the Terra Profile page External Identities tab "
                                 + "to link your account for this provider.",
                             userId, provider)));
-
-    // get client registration from provider client cache
-    var clientRegistration = providerTokenClientCache.getProviderClient(provider);
-
-    // exchange refresh token for access token
-    var accessTokenResponse =
-        oAuth2Service.authorizeWithRefreshToken(
-            clientRegistration, new OAuth2RefreshToken(linkedAccount.getRefreshToken(), null));
-
-    // save the linked account with the new refresh token to replace the old one
-    var refreshToken = accessTokenResponse.getRefreshToken();
-    if (refreshToken != null) {
-      linkedAccountService.upsertLinkedAccount(
-          linkedAccount.withRefreshToken(refreshToken.getTokenValue()));
-    }
-
-    logGetProviderAccessToken(linkedAccount, auditLogEventBuilder);
-
-    return Optional.of(accessTokenResponse.getAccessToken().getTokenValue());
```

Comment:
This was all moved to `AccessTokenCacheService` to contain all the access token logic in one place. This way, there only one entrypoint to get an access token, and other services, like fence key retrieval, can use it.